flink/v1.14/flink/src/test/java/org/apache/iceberg/flink/TestChangeLogTable.java

```diff
@@ -235,17 +236,59 @@ public void testPureInsertOnIdKey() throws Exception {
         )
     );
 
-    testSqlChangeLog(TABLE_NAME, ImmutableList.of("data"), elementsPerCheckpoint, expectedRecords);
+    testSqlChangeLog(TABLE_NAME, ImmutableList.of("data"), elementsPerCheckpoint, expectedRecords, false);
+  }
+
+  @Test
+  public void testUpsertOnIdKey() throws Exception {
```

Comment:
What happens when the table doesn't have its identifier field IDs set?

Reply:
In `FlinkSink` class, `checkAndGetEqualityFieldIds` validates the equality fields. It throws an exception when building an upsert sink without being given a primary key.